### ISSUE TITLE
CupertinoListSection should have option to remove/customize padding

### ISSUE THREAD

**Issue by Piinks** [flutter/flutter#121543]
There is padding built around the actual section which is not configurable by the user.
See image:

![Screenshot 2023-02-27 at 2 40 21 PM](https://user-images.githubusercontent.com/16964204/221680897-8f3e2314-5a0e-4a22-97aa-b056b55b469f.png)

Here it is not possible to match native behavior because of the built in leading padding. 

**Alternatively, it would probably be worth considering removing the padding all together if just wrapping CupertinoListSection in Padding achieves the same effect. This is preferred over enumerating `padding` on every widget in the framework.**

**Comment by Piinks** [comment]
See https://github.com/timsneath/landmarks-flutter/issues/13 for more context

**Comment by arrrtem22** [comment]
Can I try to fix this issue?

**Comment by MitchellGoodwin** [comment]
@arrrtem22 please do if you have the time 😃. This issue seems to be related to #128419 and both could probably be solved in one PR.

**Comment by arrrtem22** [comment]
@MitchellGoodwin I was laid off, so yeah, now I have the time😂
will you assign the issue to me?

**Comment by MitchellGoodwin** [comment]
@Piinks it looks like the padding above `CupertinoListSection` can be controlled with the `topMargin` property and can be set to 0. Which seems a little different from how we normally control padding. 
<img width="1184" alt="Screenshot 2023-06-28 at 10 07 57 AM" src="https://github.com/flutter/flutter/assets/58190796/b198b470-af17-4a8f-a79b-549aeffbfd43">
Is that enough to functionally address this issue? And should it be renamed/ changed in some way to be more clear?

**Comment by Piinks** [comment]
Looks like topMargin is exposed on the insetGrouped constructor, but is ignored? Here is the code:

```dart
import 'package:flutter/cupertino.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({super.key});

  // This widget is the root of your application.
  @override
  Widget build(BuildContext context) {
    return CupertinoApp(
      title: 'Flutter Demo',
      home: LandmarksPage(),
    );
  }
}

class LandmarksPage extends StatelessWidget {
  final List<String> landmarks = [
    "The Leaning Tower of Pisa", "The Eiffel Tower", "The Great Wall of China", "The Taj Mahal", "The Colosseum", "Machu Picchu", "The Statue of Liberty", "The Grand Canyon", "The Pyramids of Giza"
  ];

  LandmarksPage({ super.key });

  @override
  Widget build(BuildContext context) {
    return CupertinoPageScaffold(
      child: CustomScrollView(
        semanticChildCount: landmarks.length,
        slivers: [
          const CupertinoSliverNavigationBar(
            stretch: true,
            border: null,
            largeTitle: Text('Landmarks'),
            backgroundColor: CupertinoColors.systemGroupedBackground,
          ),
          SliverFillRemaining(
            hasScrollBody: false,
            child: CupertinoListSection.insetGrouped(
              children: [
                for (final landmark in landmarks) Text(landmark),
              ],
              topMargin: 0,
            ),
          ),
        ],
      ),
    );
  }
}
```

**Comment by MitchellGoodwin** [comment]
Looks like `topMargin` is ignored for the insetGrouped constructor, but `margin` does work for customizing the padding for both constructors, but it isn't clear from the documentation. For the base constructor, if both `topMargin` and a `margin` that affects the top are used, they will compound on each other, so having both seems unnecessary. It seems like the best thing to do is to remove `topMargin` and provide better documentation around `margin`.

**Comment by arrrtem22** [comment]
I agree with your observations and suggestions regarding the use of `topMargin` and margin in the constructors. Clarifying and focusing on `margin` in the documentation would be beneficial.

Also, I suggest close https://github.com/flutter/flutter/issues/128419, it would be easier for me to fix these bugs separately

**Comment by MitchellGoodwin** [comment]
Fixed by #137039. The default padding is removed, and can be added back by wrapping the header with a custom padding.
